File: backend/app/schemas/application.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

ApplicationStatus = Literal[
    "applied",
    "phone_screen",
    "interview",
    "rejected",
    "offer",
]


class ApplicationCreate(BaseModel):
    user_id: UUID
    job_id: UUID
    status: ApplicationStatus = "applied"
    notes: str | None = None


class ApplicationUpdate(BaseModel):
    status: ApplicationStatus | None = None
    notes: str | None = None


class ApplicationResponse(BaseModel):
    id: UUID
    user_id: UUID
    job_id: UUID
    status: ApplicationStatus
    applied_at: datetime
    updated_at: datetime
    notes: str | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/application.ts
export type ApplicationStatus =
  | "applied"
  | "phone_screen"
  | "interview"
  | "rejected"
  | "offer";

export type Application = {
  id: string;
  user_id: string;
  job_id: string;
  status: ApplicationStatus;
  applied_at: string;
  updated_at: string;
  notes?: string | null;
};

